snuba/clickhouse/native.py

```diff
@@ -222,15 +222,13 @@ def execute(
         query: ClickhouseQuery,
         # TODO: move Clickhouse specific arguments into DictClickhouseQuery
         settings: Optional[Mapping[str, str]] = None,
-        query_id: Optional[str] = None,
         with_totals: bool = False,
     ) -> Result:
-        if settings is None:
-            settings = {}
+        settings = {**settings} if settings is not None else {}
```

Comment:
This requires a copy since the query settings are recorded in full with `update_query_metadata_and_stats` in `snuba.web.db_query`. If this was mutable, the `pop` below would discard that value.